File: modules/coupon-domain/src/main/java/com/thanh0x/coursedeal/repository/PasskeyCredentialRepository.java
package com.thanh0x.coursedeal.repository;

import com.thanh0x.coursedeal.model.user.PasskeyCredential;
import com.thanh0x.coursedeal.model.user.UserEntity;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Optional;

@Repository
public interface PasskeyCredentialRepository extends JpaRepository<PasskeyCredential, Long> {
    Optional<PasskeyCredential> findByCredentialId(byte[] credentialId);
    List<PasskeyCredential> findAllByUser(UserEntity user);
}

File: modules/coupon-domain/src/main/kotlin/com/thanh0x/coursedeal/repository/PasskeyCredentialRepository.kt
package com.thanh0x.coursedeal.repository

import com.thanh0x.coursedeal.model.user.PasskeyCredential
import com.thanh0x.coursedeal.model.user.UserEntity
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface PasskeyCredentialRepository : JpaRepository<PasskeyCredential, Long> {
    fun findByCredentialId(credentialId: ByteArray): PasskeyCredential?
    fun findAllByUser(user: UserEntity): List<PasskeyCredential>
}
